kart/status.py

```diff
@@ -78,9 +78,16 @@ def get_type_counts(self):
     type=click.Choice(["text", "json"]),
     default="text",
 )
-def status(ctx, output_format):
+@click.option(
+    "--list-untracked-tables",
+    is_flag=True,
+    help="Shows which tables haven't yet been tracked by Kart"
+)
+def status(ctx, output_format, list_untracked_tables):
     """Show the working copy status"""
     repo = ctx.obj.get_repo(allowed_states=KartRepoState.ALL_STATES)
+    untracked_tables = []
+    is_untracked_tables_call = False
```

Comment:
Hm, I did this to separate 4 scenarios:

Calling --untracked_tables while having untracked tables
Calling --untracked_tables while not having untracked tables
Not calling --untracked_tables (only calling `kart status`) while having untracked tables
Not calling --untracked_tables  while not having untracked tables

If only checking the presence of untracked tables, it will append the output with the result of this check even when calling only `kart status`

Reply:
I understand the four separate scenarios - I will address them in a moment -  but this variable is still a duplicate: you have a variable `list_untracked_tables` and another variable `is_untracked_tables_call`. You copy the value of `list_untracked_tables` onto `is_untracked_tables_call`. Anything you use `is_untracked_tables_call` you could more simply use `list_untracked_tables` for to avoid creating a duplicate variable.

Regarding separating the four scenarios. As noted above, you can resolve them without creating a duplicate variable - but I propose the following logic, which I think is simpler:
1. When generating JSON, you wire through the `list_untracked_tables` variable so you know whether you should generate the list of untrackd tables or not.
2. If it is True, you set the status["workingCopy"]["untrackedTables"] field. If there are untracked tables, you set it to a list of those, if there are none, you set it to an empty list.
3. When converting the JSON to text, you don't need to wire through any variables at all.
Instead, you test for the presence of the "untrackedTables" key. You can do this with something like `if "untrackedTables" in some_dict:`
4. If it is not present, you don't output anything about untracked tables. If it is present, you check if it is empty or not in order to decide what to output.

(Note that there are really only 3 separate scenarios since, if the user doesn't call --untracked-tables, we don't actually know if there are untracked tables or not)